benchmark/src/submit_pattern.rs

```diff
@@ -0,0 +1,100 @@
+use logger::prelude::*;
+use std::{thread, time, vec::IntoIter};
+
+/// For constant rate X items per second,
+/// ConstantRate iterator returns 1 item and then sleep for SUBMISSION_TIME_UNIT_US / X us.
+const SUBMISSION_TIME_UNIT_US: u64 = 1_000_000;
```

Comment:
This is microseconds in 1 second, and to satisfy the constant rate requirement, this cannot be anything else -- it's not something to be adjustable. Can you rename it, or directly use micros in second from duration?

Reply:
Perhaps my naming/comment here and for `constant` are still not clear. This unit can be something else, as long as it is consistent with the unit of `constant` in `ConstantRate`. For example, if `constant` stands for X items per 10 milliseconds, then `SUBMISSION_TIME_UNIT_US` should be 10_000 microseconds.

However, using "items per second" is usually most straightforward to user.